ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -890,9 +915,7 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         ORKTableSection *section = _sections[indexPath.section];
         ORKTableCellItem *cellItem = section.items[indexPath.row - 1];
         [section.textChoiceCellGroup didSelectCellAtIndexPath:indexPath];
-        
-        id answer = (cellItem.formItem.questionType == ORKQuestionTypeBoolean)? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
-        
+        id answer = ([cellItem.formItem.answerFormat isKindOfClass:[ORKBooleanAnswerFormat class]])? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
```

Comment:
@umerkhan-apple Yep it is a bug. I guess nobody noticed it so far, since `ORKBooleanQuestionResult`'s `setAnswer` method checks if there is a single boolean value wrapped in an array (and unwraps it if necessary) before actually changing the `booleanAnswer` property.